walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -24,33 +24,28 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    // TODO: Maybe use binary tree or another data structure for availableTimes 
-    // because branches split in two?
-    // TODO: Combine the repetition of calling getAvailableTimes() below so it
-    // doesn't need to repeat without optional attendees.
     Collection<TimeRange> availableTimes = new ArrayList<>();
+    Collection<TimeRange> availableTimesWithoutOptional;
     Collection<String> attendees = request.getAttendees();
     Collection<String> optionalAttendees = request.getOptionalAttendees();
-    Collection<String> allAttendees = new ArrayList<>(request.getAttendees());
-    allAttendees.addAll(optionalAttendees);
     long duration = request.getDuration();
     if (duration > TimeRange.WHOLE_DAY.duration()){
       return availableTimes;
     }
-    availableTimes = getAvailableTimes(events, allAttendees, duration);
-    if (availableTimes.isEmpty()){
-      availableTimes = getAvailableTimes(events, attendees, duration);
+    availableTimes.add(TimeRange.WHOLE_DAY);
+    getAvailableTimes(availableTimes, events, attendees, duration);
```

Comment:
Conceptually what's the difference between these two methods?  The `getAvailableTimes` method evaluates time slots and removes them from the available list.  The `getAvailableTimesWithMaxOptional` method also evaluates time slots, but instead of removing from the available list it scores them.  But at the very core, they both evaluate the time slot and returns a decision to either remove or keep+rank.